code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -99,6 +101,23 @@
 	stuck_to = hit_atom
 	RegisterSignal(stuck_to, COMSIG_QDELETING, PROC_REF(clean_refs))
 
+/obj/item/explosive/grenade/sticky/afterattack(atom/target, mob/user, has_proximity, click_parameters)
+	. = ..()
+	if(target != user)
+		return
+	if(self_sticky == FALSE)
+		return
+	user.drop_held_item()
+	activate()
+	var/image/stuck_overlay = image(icon, user, initial(icon_state) + "_stuck")
+	stuck_overlay.pixel_x = rand(-5, 5)
+	stuck_overlay.pixel_y = rand(-7, 7)
+	user.add_overlay(stuck_overlay)
+	forceMove(user)
+	saved_overlay = stuck_overlay
+	stuck_to = user
+	RegisterSignal(stuck_to, COMSIG_QDELETING, PROC_REF(clean_refs))
```

Comment:
You're still copy pasting a lot.
This whole thing should just be some 'stuck_to' proc or something, then just add to that as required for the child types.